sensor/common/resources/namespaces/namespace_store.go

```diff
@@ -0,0 +1,65 @@
+package namespaces
+
+import (
+	"testing"
+
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/sync"
+)
+
+// NamespaceStore is a namespace store stores a mapping of namespace names to their ids.
+type NamespaceStore struct {
+	lock sync.RWMutex
+
+	namespaceNamesToMetadata map[string]namespaceMetadata
+}
+
+type namespaceMetadata struct {
+	id         string
+	annotation map[string]string
+}
+
+func newNamespaceStore() *NamespaceStore {
+	return &NamespaceStore{
+		namespaceNamesToMetadata: make(map[string]namespaceMetadata),
+	}
+}
+
+// NewTestNamespaceStore returns a namespace store for testing purposes
+func NewTestNamespaceStore(t *testing.T) *NamespaceStore {
+	if t == nil {
+		return nil
+	}
+	return newNamespaceStore()
+}
+
+// AddNamespace adds a namespace to the datastore.
+func (n *NamespaceStore) AddNamespace(ns *storage.NamespaceMetadata) {
+	n.lock.Lock()
+	defer n.lock.Unlock()
+
+	n.namespaceNamesToMetadata[ns.GetName()] = namespaceMetadata{
+		id:         ns.GetId(),
+		annotation: ns.GetAnnotations(),
+	}
+}
+
+// GetAnnotationsForNamespace returns the annotations for the given namespace.
+func (n *NamespaceStore) GetAnnotationsForNamespace(name string) map[string]string {
+	n.lock.RLock()
+	defer n.lock.RUnlock()
+
+	return n.namespaceNamesToMetadata[name].annotation
+}
+
+// LookupNamespaceID returns the ID of a given namespace if it exists.
+func (n *NamespaceStore) LookupNamespaceID(name string) (string, bool) {
```

Comment:
LookupNamespaceName

Reply:
I'm confused by this comment, the function takes a namespace name and returns the ID of the namespace in question. Seems like it's named properly to me. I would also say that this was how it was named already

Reply:
Correct, in that it was already named that way. The func takes a namespace by name and looks it up in a map keyed by name, hence the suggestion.